File: app/src/main/java/com/ireddragonicy/konabessnext/utils/ThreadUtil.java
package com.ireddragonicy.konabessnext.utils;

import android.os.Handler;
import android.os.Looper;
import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

public class ThreadUtil {
    private static final ExecutorService executor = Executors.newCachedThreadPool();
    private static final Handler mainHandler = new Handler(Looper.getMainLooper());

    public static void runInBackground(Runnable runnable) {
        executor.execute(runnable);
    }

    public static void runOnMain(Runnable runnable) {
        if (Looper.myLooper() == Looper.getMainLooper()) {
            runnable.run();
        } else {
            mainHandler.post(runnable);
        }
    }
    
    public static void runOnMainDelayed(Runnable runnable, long delayMillis) {
        mainHandler.postDelayed(runnable, delayMillis);
    }
}

File: app/src/main/java/com/ireddragonicy/konabessnext/utils/ThreadUtil.kt
package com.ireddragonicy.konabessnext.utils

import android.os.Handler
import android.os.Looper
import java.util.concurrent.Executors

object ThreadUtil {
    private val executor = Executors.newCachedThreadPool()
    private val mainHandler = Handler(Looper.getMainLooper())

    @JvmStatic
    fun runInBackground(runnable: Runnable) {
        executor.execute(runnable)
    }
    
    // Kotlin-friendly overload
    fun runInBackground(action: () -> Unit) {
        executor.execute(action)
    }

    @JvmStatic
    fun runOnMain(runnable: Runnable) {
        if (Looper.myLooper() == Looper.getMainLooper()) {
            runnable.run()
        } else {
            mainHandler.post(runnable)
        }
    }
    
    // Kotlin-friendly overload
    fun runOnMain(action: () -> Unit) {
        if (Looper.myLooper() == Looper.getMainLooper()) {
            action()
        } else {
            mainHandler.post(action)
        }
    }

    @JvmStatic
    fun runOnMainDelayed(runnable: Runnable, delayMillis: Long) {
        mainHandler.postDelayed(runnable, delayMillis)
    }
    
    fun runOnMainDelayed(delayMillis: Long, action: () -> Unit) {
        mainHandler.postDelayed(action, delayMillis)
    }
}
